ui/lib/sync/addon/components/secrets/sync-activation-modal.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { service } from '@ember/service';
+import { task } from 'ember-concurrency';
+import { waitFor } from '@ember/test-waiters';
+import errorMessage from 'vault/utils/error-message';
+
+import type FlashMessageService from 'vault/services/flash-messages';
+import type StoreService from 'vault/services/store';
+import type RouterService from '@ember/routing/router-service';
+
+interface Args {
+  onClose: () => void;
+  onError: (errorMessage: string) => void;
+  isHvdManaged: boolean;
+}
+
+export default class SyncActivationModal extends Component<Args> {
+  @service declare readonly flashMessages: FlashMessageService;
+  @service declare readonly store: StoreService;
+  @service declare readonly router: RouterService;
+
+  @tracked hasConfirmedDocs = false;
```

Comment:
since we don't reset this tracked property on cancel anymore, does it stay checked if a user clicks `Cancel` ?

Reply:
it does not stay checked. 🙂 the instance of `<Secrets::SyncActivationModal />` is unmounted and destroyed when the parent component's `this.showActivateSecretsSyncModal` tracked property changes, as shown here:

https://github.com/hashicorp/vault/blob/dcb6b1a0721ce3674195bca4513b4aa123e95456/ui/lib/sync/addon/components/secrets/page/overview.hbs#L203-L209 

once `this.showActivateSecretsSyncModal` is set to true again, a new instance of `<Secrets::SyncActivationModal />` is rendered that does not retain any state from the previous state. 

i think testing this scenario feels like testing Ember internals. if the conditional to show/hide the modal was contained inside the modal component itself, i'd say we should definitely test it, but the perk of us having the conditional in the parent component is that we can rely on Ember to know when to tear down / set up new component instances. 

how does that sound to you? 😀

Reply:
Ah yeah - I forgot that HDS modals _actually_ leave the DOM....still recovering from ember wormhole days where they were permanently rendered 🤯